### PR TITLE
UserDB and UserBuilder module tests

### PR DESCRIPTION
**Pull request by arcturus**
@ferm r?

### CODE REVIEW

File: src/users_db.rs [hunk 1]
```diff
@@ -151,3 +151,86 @@ impl UsersDb {
...
+
+        assert_eq!(user.id, Some(1));
+        assert_eq!(user.name, "Mr Fox");
+        assert_eq!(user.email, "fox@mozilla.org");
+        let mut md5 = Md5::new();
+        md5.input_str("pass12345678");
+        assert_eq!(user.password, md5.result_str());
+    }
+
+    failing "should panic if invalid user" {
```
Comment (ferjm) [inline src/users_db.rs:177]: We would need to test the different failure reasons: empty or short password, empty name, empty email, malformed email. But that can be done in a follow up.

**Comment by ferjm** [comment]
Thanks!

File: src/users_db.rs (hunk 1 above)
Comment (arcturus) [inline src/users_db.rs:177]: Yes, I was thinking about that, but got lazy.

Promise doing in a follow up.
